apps/api/routes.py

```diff
@@ -505,3 +505,181 @@ def extend_lab(lab_id):
     )
 
     return {"status": "ok", "result": new_expiration}, 200
+
+@blueprint.route("/namespaces", methods=["POST"])
+@login_required
+def create_namespace():
+    if current_user.category not in ["admin", "teacher"]:
+        return {"status": "fail", "result": "Unauthorized access."}, 401
+
+    content = request.get_json(silent=True)
+    if not content:
+        return {"status": "fail", "result": "Invalid content."}, 400
+
+    if current_user.category == "admin":
+        content["is_approved"] = True
+        content["is_enabled"] = True
+        
+    owners = [int(x) for x in json.loads(content.get("owners", "[]"))]
+    members = [int(x) for x in json.loads(content.get("members", "[]"))]
+
+    if current_user.id not in owners:
+        owners = json.dumps([current_user.id] + owners)
+        
+    content["owners"] = json.dumps(list(set(owners)))
+    content["members"] = json.dumps(list(set(members)))
+    content["user_id"] = current_user.id
+
+    namespace = Namespaces(**content)
+    db.session.add(namespace)
+    db.session.commit()
+
+    return {"status": "ok", "result": "Namespace saved successfully"}, 200
+
+
+@blueprint.route("/namespaces/<namespace_id>", methods=["GET"])
+@login_required
+def get_namespace_by_id(namespace_id):
+    namespace = Namespaces.query.get(namespace_id)
+    if not namespace:
+        return {"status": "fail", "result": "Namespace not found"}, 404
+
+    users_in_namespace = json.loads(namespace.owners or "[]") + json.loads(namespace.members or "[]")
+    if current_user.id not in users_in_namespace:
+        return {"status": "fail", "result": "Unauthorized access to this namespace."}, 401
+
+    return {
+        "status": "ok",
+        "result": namespace.as_dict()
+    }, 200
+
+
+@blueprint.route("/namespaces/<namespace_id>", methods=["PUT"])
+@login_required
+def update_namespace(namespace_id):
+    if current_user.category not in ["admin", "teacher"]:
+        return {"status": "fail", "result": "Unauthorized access"}, 401
```

Comment:
we have a decorator called "check_user_category()" that can be used to reduce code duplicated like this validation.